### PR TITLE
Fix role mapping add, update, delete flows

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose
- Fix role mapping not updating issues

### Related Issues
- https://github.com/wso2/product-is/issues/13482

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by brionmario — APPROVED** [review]

File: modules/react-components/src/components/input/dynamic-field.tsx [hunk 1]
```diff
@@ -167,6 +167,9 @@ export const DynamicField: FunctionComponent<DynamicFieldPropsInterface> = (
      * Prevent submit from being triggered during initial render
      */
     useEffect(() => {
+        if(!update) {
```
Comment (Yasara-Y) [inline modules/react-components/src/components/input/dynamic-field.tsx:170]: Shall we fix formatting here?

File: apps/console/src/features/applications/components/settings/attribute-management/role-mapping.tsx [hunk 2]
```diff
@@ -145,23 +151,20 @@ export const RoleMapping: FunctionComponent<RoleMappingPropsInterface> = (
...
                                 ".applicationRole.validations.duplicate")
                         }
-                        submit={ submitState }
-                        update={ (data) => {
-                            if (data.length > 0) {
-                                const finalData: RoleMappingInterface[] = data.map(mapping => {
+                        readOnly={ readOnly }
+                        data-testid={ `${ testId }-dynamic-field` } 
+                        listen={ (data) => {
+                            if(onChange) {
```
Comment (Yasara-Y) [inline apps/console/src/features/applications/components/settings/attribute-management/role-mapping.tsx:157]: Lets fix formatting here.

Comment (JayaShakthi97) [reply apps/console/src/features/applications/components/settings/attribute-management/role-mapping.tsx:157]: Fixed with 8535eecb987fefadc2bc7058e3a9070e260baf10

File: modules/react-components/src/components/input/dynamic-field.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply modules/react-components/src/components/input/dynamic-field.tsx:170]: Fixed with 8535eecb987fefadc2bc7058e3a9070e260baf10

**Pull request merged**